src/pepr/operator/controllers/istio/injection.ts

```diff
@@ -88,7 +93,10 @@ export async function cleanupNamespace(pkg: UDSPackage) {
     { force: true },
   );
 
-  await killPods(pkg.metadata.namespace, false);
+  // Kill the pods if we changed the value of the istio-injection label
+  if (originalInjectionLabel !== labels[injectionLabel]) {
+    await killPods(pkg.metadata.namespace, false);
+  }
```

Comment:
I think this works and probably makes the most sense to handle the case where the original injection label was already "enabled". Noting mostly for myself, but the problem here was this not having the same `If there are no more UDS Package annotations` check around it (although your check handles everything best I think).